podman/tests/integration/test_containers.py

```diff
@@ -37,17 +51,87 @@ def test_container_crud(self):
             self.assertIsInstance(actual, Container)
             self.assertEqual(actual.id, container.id)
 
+        file_contents = b"This is an integration test for archive."
+        file_buffer = io.BytesIO(file_contents)
+
+        tar_buffer = io.BytesIO()
+        with tarfile.open(fileobj=tar_buffer, mode="w") as tar:
+            info = tarfile.TarInfo("/")
+            info.type = tarfile.DIRTYPE
+            tar.addfile(info)
+
+            info = tarfile.TarInfo("unittest")
+            info.size = len(file_buffer.getbuffer())
+            tar.addfile(info, file_buffer)
+            tarball = tar_buffer.getvalue()
+
+        with self.subTest("Archive /data/unittest"):
+            self.assertTrue(container.put_archive("/data", data=tarball))
+
+            actual, stats = container.get_archive("/data")
+
+            with io.BytesIO() as fd:
+                for chunk in actual:
+                    fd.write(chunk)
+                fd.seek(0, 0)
+
+                self.assertTrue(tarfile.is_tarfile(name=fd))
+                with tarfile.open(fileobj=fd, mode="r") as tar:
+                    contents = tar.extractfile("data/unittest").read()
+                self.assertEqual(contents, file_contents)
+
         with self.subTest("List containers --all"):
             containers = self.client.containers.list(all=True)
             self.assertGreater(len(containers), 0)
             ids = [i.id for i in containers]
             self.assertIn(container.id, ids)
 
+        with self.subTest("Get container's logs"):
+            container.start()
+            container.wait(condition="exited")
+
+            logs_iter = container.logs(stream=False)
+            self.assertIsInstance(logs_iter, Iterator)
+
+            logs = list(logs_iter)
+            self.assertIn(random_string.encode("utf-8"), logs)
+
         with self.subTest("Delete Container"):
             container.remove()
             with self.assertRaises(NotFound):
                 self.client.containers.get(container.id)
 
+        with self.subTest("Run Container"):
+            top_ctnr = self.client.containers.run(
+                self.alpine_image, "/usr/bin/top", name="TestRunPs", detach=True
+            )
+            self.assertEqual(top_ctnr.status, "running")
+
+            top_ctnr.pause()
+            top_ctnr.reload()
+            self.assertEqual(top_ctnr.status, "paused")
+
+            top_ctnr.unpause()
+            top_ctnr.reload()
+            self.assertEqual(top_ctnr.status, "running")
+
+            report = top_ctnr.top()
+            self.assertIn("/usr/bin/top", report["Processes"][0])
```

Comment:
[From the test failure](https://cirrus-ci.com/task/6530551025762304?logs=main#L3675), it appears this needs to be `self.assertIn("/usr/bin/top", report["Processes"][0].strip())`